packages/grafana-data/src/valueFormats/valueFormats.ts

```diff
@@ -45,10 +45,15 @@ export function toFixed(value: number, decimals?: DecimalCount): string {
   if (value === null) {
     return '';
   }
+
   if (value === Number.NEGATIVE_INFINITY || value === Number.POSITIVE_INFINITY) {
     return value.toLocaleString();
   }
 
+  if (decimals === null || decimals === undefined) {
+    decimals = getDecimalsForValue(value);
+  }
```

Comment:
This is the important part, after changing so auto decimals are not assigned in the displayProcessor before the format function is called and not in toFixedScaled it comes in as null or undefined here, but the value is scaled and it is now easier to figure out the auto decimals!